proseco/tests/test_acq.py

```diff
@@ -588,6 +589,35 @@ def test_box_strategy_20603(proseco_agasc_1p7):
     assert repr(acqs[TEST_COLS]).splitlines() == exp
 
 
+def test_overlap_penalty():
+    # Default setting which is to have overlap penalty
+    stars = StarsTable.empty()
+
+    stars.add_fake_constellation(
+        mag=[7.0, 7.1, 7.2, 7.3, 7.4, 7.5],
+        size=2000,
+        n_stars=6,
+    )
+    stars.add_fake_constellation(
+        mag=[9.0, 9.1],
+        size=1000,
+        n_stars=2,
+    )
+    stars.add_fake_star(yang=stars["yang"][0] - 240, zang=stars["zang"][0], mag=7.4)
+    stars.pprint_all()
+    with ska_helpers.utils.temp_env_var("PROSECO_DISABLE_OVERLAP_PENALTY", "False"):
```

Comment:
Should this be monkeypatch.setenv for the test case use?  I figured temp_env_var was the right thing for non-test code but it probably doesn't matter.

Reply:
This test needs to set the env within a specific context. AFAIK `monkeypatch` just works globally following the command so it is not quite what we want here.

Also a gentle nudge to lower the bar on bringing up style nitpicks especially in test code.

Reply:
Good point about the context, thanks for the reminder!  We'd probably need monkeypatch.context() but this use case makes sense to me now for temp_env_var.  And understood about style nitpicks -- this mostly raised a flag for me because not using monkeypatch felt similar to the previous issue we had with testing env vars indeed creeping out of the individual test scope and causing test side effect -- but by explicitly using the context manager it isn't an issue here.

Reply:
I didn't know about the `.context()` method, which would work here. Maybe next time if I remember by then.